src/lightning_app/utilities/proxies.py

```diff
@@ -450,6 +454,8 @@ def _sigterm_signal_handler(self, signum, frame, call_hash: str) -> None:
         """Signal handler used to react when spot instances are being retrived."""
         logger.info(f"Received SIGTERM signal. Gracefully terminating {self.work.name.replace('root.', '')}...")
         persist_artifacts(work=self.work)
+        # TODO: Introduce cleaner hooks.
+        self.work.on_exit()
```

Comment:
Are you looking for something more granular? Or a better name, or some specific hook just for this call here? I can help with this, we can discuss later.

Reply:
Sounds good. Any suggestions ?

Reply:
We decided to remove it from the runtime and keep it within the proxy.